### ISSUE TITLE
Get-PnPProvisioningTemplate : Data at the root level is invalid

### ISSUE THREAD

**Issue by mkorek** [SharePoint/PnP-Sites-Core#2247]
#### Category
[x] Bug
[ ] Enhancement

#### Environment
[x] Office 365 / SharePoint Online
[ ] SharePoint 2016
[ ] SharePoint 2013

Just updated to 3.9.1905.2

#### Expected or Desired Behavior
Being able to generate a template for a specific site

When trying to generate a template via PowerShell for a site I get the following error:
_Get-PnPProvisioningTemplate : Data at the root level is invalid. Line 1, position 1._

So I enabled logging and retried generating a template for the site and found that it fails at the very last step where it would spit out the pnp file.  So no template file gets generated.
 
To troubleshoot, I first tried generating a template of a different site…which worked.  Knowing that it was an issue with the original site, I stepped through each of the handlers to see which one generated the error.  By stepping through I mean, I did Get-PnPProvisioningTemplate -path $blah -handler (each one separately) and found that the error only occurred when the list handler was included.

I know this is an issue on my end, but I honestly have no idea where to go from here.  I was able to generate a template for the site with the previous version on pnp I was on (3.4.1812.2), but needed to update to the latest version due to the content type issue.

Any input would be beyond appreciated!

Matt

**Comment by mkorek** [comment]
if it helps, this is the stacktrace:

$error[0].Exception.Stacktrace
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.ParseRootLevelWhitespace()
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.XmlReader.MoveToContent()
   at System.Xml.Linq.XElement.Load(XmlReader reader, LoadOptions options)
   at System.Xml.Linq.XElement.Parse(String text, LoadOptions options)
   at OfficeDevPnP.Core.Framework.Provisioning.Providers.Xml.Resolvers.ListViewsFromModelToSchemaTypeResolver.Resolve(Object source, Dictionary`2 resolvers, Boolean recursive)
   at OfficeDevPnP.Core.Framework.Provisioning.Providers.Xml.PnPObjectsMapper.MapProperties(Object source, Object destination, Dictionary`2 resolvers, Boolean recursive)
   at OfficeDevPnP.Core.Framework.Provisioning.Providers.Xml.Resolvers.CollectionFromModelToSchemaTypeResolver.Resolve(Object source, Dictionary`2 resolvers, Boolean recursive)
   at OfficeDevPnP.Core.Framework.Provisioning.Providers.Xml.Serializers.V201705.ListInstancesSerializer.Serialize(ProvisioningTemplate template, Object persistence)
   at OfficeDevPnP.Core.Framework.Provisioning.Providers.Xml.XmlPnPSchemaBaseSerializer`1.InvokeSerializers(ProvisioningTemplate template, Object persistenceTemplate, IOrderedEnu
merable`1 serializers, SerializationAction action)
   at OfficeDevPnP.Core.Framework.Provisioning.Providers.Xml.XmlPnPSchemaBaseSerializer`1.SerializeTemplate(ProvisioningTemplate template, Object persistenceTemplate)
   at OfficeDevPnP.Core.Framework.Provisioning.Providers.Xml.XmlPnPSchemaBaseSerializer`1.ProcessOutputStream(ProvisioningTemplate template, TSchemaTemplate result)
   at OfficeDevPnP.Core.Framework.Provisioning.Providers.Xml.XmlPnPSchemaBaseSerializer`1.ToFormattedTemplate(ProvisioningTemplate template)
   at OfficeDevPnP.Core.Framework.Provisioning.Providers.TemplateProviderBase.SaveToConnector(ProvisioningTemplate template, String uri, ITemplateFormatter formatter, ITemplatePr
oviderExtension[] extensions)
   at SharePointPnP.PowerShell.Commands.Provisioning.Site.GetProvisioningTemplate.ExtractTemplate(XMLPnPSchemaVersion schema, String path, String packageName)
   at SharePointPnP.PowerShell.Commands.Provisioning.Site.GetProvisioningTemplate.ExecuteCmdlet()
   at SharePointPnP.PowerShell.Commands.PnPCmdlet.ProcessRecord()

**Comment by jansenbe** [comment]
Are you using a particular schema version at extraction time? Can you share the command line you're using?

**Comment by mkorek** [comment]
Hello, I'm using:
Get-PnPProvisioningTemplate -path "C:\path\model.pnp"

**Comment by jackpoz** [comment]
I had the same issue, this is because the ServerRelativeUrl in case of the root site collection is / and PnP replaces all the / occurrences in the schema with {site} placeholder. 
If I inspect the content of https://github.com/SharePoint/PnP-Sites-Core/blob/master/Core/OfficeDevPnP.Core/Framework/Provisioning/Providers/Xml/Resolvers/ListViewsFromModelToSchemaTypeResolver.cs#L38 , this is what I get
```xml
<View Name="{F96F0EF4-71FD-46D6-9456-AD4796727EA5}" DefaultView="TRUE" MobileView="TRUE" MobileDefaultView="TRUE" Type="HTML" DisplayName="All items" Url="{site}Lists{site}Announcement{site}AllItems.aspx" Level="1" BaseViewID="1" ContentTypeID="0x" ImageUrl="{site}_layouts{site}15{site}images{site}announce.png?rev=47">
  <Query>
    <OrderBy>
      <FieldRef Name="Modified" Ascending="FALSE" {site}>
    <{site}OrderBy>
  <{site}Query>
  <ViewFields>
    <FieldRef Name="LinkTitle" {site}>
    <FieldRef Name="Modified" {site}>
  <{site}ViewFields>
  <RowLimit Paged="TRUE">30<{site}RowLimit>
  <JSLink>clienttemplates.js<{site}JSLink>
<{site}View>
```

![image](https://user-images.githubusercontent.com/1153754/59104924-1af34a00-8933-11e9-978d-f2ef63f97c3d.png)

**Comment by mkorek** [comment]
@jackpoz thank you very much for the reply!

Does this happen for you when you try to extract any list?  Or only for lists with JSLink utilized in the views?    

The only reason I ask is because, when I was originally testing this, the site I was able to successfully generate a template for didn't utilize JSLink in any of the list views.  While the site that was failing, did.

Not sure if that helps at all but figured I'd throw it out there in case it did.  Thanks again!

**Comment by jackpoz** [comment]
@mkorek this issue happens even when creating a new demo tenant and calling "Get-PnPProvisioningTemplate -Out testroot.xml -Handlers Lists" without creating any custom list.

I created a pull request https://github.com/SharePoint/PnP-Sites-Core/pull/2295 , give it a try :)

**Comment by mkorek** [comment]
@jackpoz thanks again for the reply! I'll try it out later today!

**Comment by mkorek** [comment]
@jackpoz  sorry for the delay in response!  I just tried updating pnp back to the version I was getting the error on (3.9.1905.2) and I don't seem to be getting the error anymore : / so I don't really have anything to test your pr against.  

The only thing I can really think of is that I've gotten a new computer since I reported the issue, and maybe it was being caused by conflicting modules on my old computer.  Not really sure, but this is kind of a new realm for me.

Thanks again for all your help though!
